sources/packages/api/src/cra-integration/cra-integration.service.ts

```diff
@@ -182,104 +147,51 @@ export class CRAIntegrationService {
-      const fileLines = fileContent
-        .toString()
-        .split(/\r\n|\n\r|\n|\r/)
-        .filter((line) => line.length > 0);
-      // Read the first line to check if the header code is the expected one.
-      const header = CRAFileHeader.CreateFromLine(fileLines.shift()); // Read and remove header.
-      if (header.transactionCode !== TransactionCodes.ResponseHeader) {
-        this.logger.error(
-          `The CRA file ${fileName} has an invalid transaction code on header: ${header.transactionCode}`,
-        );
-        // If the header is not the expected one, just ignore the file.
-        return null;
-      }
-
-      // Remove footer (not used).
-      fileLines.pop();
-
-      // Generate the records.
-      let lineNumber = 1;
-      const statusRecords: CRAResponseStatusRecord[] = [];
-      const totalIncomeRecords: CRAResponseTotalIncomeRecord[] = [];
-      for (const line of fileLines) {
-        const craRecord = new CRAResponseRecordIdentification(line, lineNumber);
-        switch (craRecord.transactionSubCode) {
-          case TransactionSubCodes.ResponseStatusRecord:
-            statusRecords.push(new CRAResponseStatusRecord(line, lineNumber));
-            break;
-          case TransactionSubCodes.TotalIncome:
-            totalIncomeRecords.push(
-              new CRAResponseTotalIncomeRecord(line, lineNumber),
-            );
-            break;
-        }
-        lineNumber++;
-      }
-
-      return {
-        fileName,
-        filePath,
-        statusRecords,
-        totalIncomeRecords,
-      };
-    } finally {
-      await SshService.closeQuietly(client);
+  async downloadResponseFile(
+    remoteFilePath: string,
+  ): Promise<CRAsFtpResponseFile> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    // Read the first line to check if the header code is the expected one.
+    const header = CRAFileHeader.CreateFromLine(fileLines.shift()); // Read and remove header.
+    if (header.transactionCode !== TransactionCodes.ResponseHeader) {
+      this.logger.error(
+        `The CRA file ${remoteFilePath} has an invalid transaction code on header: ${header.transactionCode}`,
+      );
+      // If the header is not the expected one, throw an error.
+      throw new Error("Invalid file header.");
     }
-  }
 
-  /**
-   * Delete a file from SFTP.
-   * @param filePath Full path of the file to be deleted.
-   */
-  async deleteFile(filePath: string): Promise<void> {
-    const client = await this.getClient();
-    try {
-      await client.delete(filePath);
-    } finally {
-      await SshService.closeQuietly(client);
+    // Remove footer (not used).
+    fileLines.pop();
+
+    // Generate the records.
+    let lineNumber = 1;
+    const statusRecords: CRAResponseStatusRecord[] = [];
+    const totalIncomeRecords: CRAResponseTotalIncomeRecord[] = [];
+    for (const line of fileLines) {
+      const craRecord = new CRAResponseRecordIdentification(line, lineNumber);
```

Comment:
Instead of declaring a new variable lineNumber, can we use line+1 e.g CRAResponseRecordIdentification(line, line+1)

Am I missing something?

Reply:
Sorry, I am not following. The variable is created to be used along the block and needs to be incremented for every loop. 
What would be the alternative for the below block?

![image](https://user-images.githubusercontent.com/61259237/141178745-e2d0f3bc-1c6f-465d-8df2-3962604991db.png)

Reply:
All good, if it simplifies the code it is always better. I believe that the final simplification would be like the below, right?
![image](https://user-images.githubusercontent.com/61259237/141180902-a99b0071-6d30-4f30-b4b4-26e17e0a938c.png)